Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/NavigationController.java

```diff
@@ -207,4 +241,68 @@ protected void share(FolderType selectedFolder) {
 			Messagebox.show(e.getMessage(), APROMORE, Messagebox.OK, Messagebox.ERROR);
 		}
 	}
+
+
+	public void rename(FolderType selectedFolder) throws InterruptedException {
+		
+			if (selectedFolder == null) {
+				Notification.error(Labels.getLabel("portal_failedFind_message"));
+				return;
+			}
+		
+		try {
+			boolean canChange = false;
+			try {
+				User currentUser = mainC.getSecurityService().getUserById(UserSessionManager.getCurrentUser().getId());
+				canChange = ItemHelpers.canModify(currentUser, selectedFolder);
+			} catch (Exception e) {
+				Notification.error(e.getMessage());
+				return;
+			}
+
+			if (canChange) {
+				mainC.eraseMessage();
+				new RenameFolderController(mainC, selectedFolder.getId(), selectedFolder.getFolderName());
+			} else {
+				Notification.error(Labels.getLabel("portal_noPrivilegeRename_message"));
+			}
+
+		} catch (DialogException e) {
+			Messagebox.show(e.getMessage(), APROMORE, Messagebox.OK, Messagebox.ERROR);
+		}
+	}
+
+
+	public void removeFolder(FolderType selectedFolder) {
+		
+		if (selectedFolder == null) {
+			Notification.error(Labels.getLabel("portal_failedFind_message"));
+			return;
+		}
+		
+		Messagebox.show(Labels.getLabel("portal_deleteFolderPrompt_message"), Labels.getLabel(PORTAL_WARNING_TEXT),
+				Messagebox.YES | Messagebox.NO, Messagebox.QUESTION, new EventListener<Event>() {
+					@Override
+					public void onEvent(Event evt) throws Exception {
+						switch (((Integer) evt.getData())) {
+						case Messagebox.YES:
+							try {
+								mainC.getManagerService().deleteFolder(selectedFolder.getId(),
+										UserSessionManager.getCurrentUser().getUsername());
+							} catch (Exception e) {
+								LOGGER.error("Failed to delete folder from Tree", e);
+								Messagebox.show(
+										"Could not perform all delete operations. You may not be authorized to delete some of the resources.",
```

Comment:
This is user-visible text, so it should probably be internationalized like the rest.